### PR TITLE
Enable authentication for ElasticSearch/OpenSearch

### LINKED ISSUE #2233: Access Elasticsearch with authentication from the Chef Server.
As a Chef Infra Server customer, during the security scan we found that "Elasticsearch Accessible Without Authentication" on port 9200. please find the details below.

Threat:
Elasticsearch is a search engine based on Lucene. It provides a distributed, multitenant-capable full-text search engine with an HTTP web interface and schema-free JSON documents.

It's possible to access the elasticsearch server without any authentication.

QID Detection Logic:
This QID sends "GET" request to the elasticsearch port to check if authentication is needed.

Impact:
Exploitation of this vulnerability allows a remote attacker to obtain sensitive information.

Suggested Solution:
Enable authentication.

Some Additional Context:

We will be moving to a supported version of OpenSearch where authentication(x-pack) will be enabled. (More discussion on OpenSearch vs OpenDistro on #2813)

Current idea:
- Enable authentication on es (https://github.com/chef/chef-server/issues/2861)
- Create an es-proxy service on chef-server. The es proxy service will have the credentials to log into es.
- Any request coming to chef-server will be routed through the es-proxy service to es.
- curl without password will error out.

Acceptance Criteria:
- This is a spike to identify the body of work that will need to happen on server side to support enabling auth in Elasticsearch.
- Create relevant tickets for the identified work.

### PR DESCRIPTION
**Pull request by jashaik**
### Description
List of itesm to be added

1. Investigate auth plugin - (should also support Opensearch)
2. Investigate omnibus changes for omnibus build
3. Investigate es_gateway to add basic authentication header
4. Investigate habitat changes for habitat build

### Issues Resolved
#2233 

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: omnibus_overrides.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@
 override :ruby, version: "2.7.4"
 override :perl, version: "5.18.1"
 override :redis, version: "5.0.7"
+override :elasticsearch, version: "7.9.1"
```
Comment (tas50) [inline omnibus_overrides.rb:12]: you wouldn't need to set this if the default was changed

**Comment by jashaik** [comment]
Pipeline Information -
Adhoc - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3315
Umbrella -

File: omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb [hunk 2]
```diff
@@ -317,6 +317,9 @@
 default['private_chef']['opscode-erchef']['search_queue_mode'] = 'batch'
 default['private_chef']['opscode-erchef']['search_batch_max_size'] = '5000000'
 default['private_chef']['opscode-erchef']['search_batch_max_wait'] = '10'
+default['private_chef']['opscode-erchef']['search_auth_enabled'] = false
+default['private_chef']['opscode-erchef']['search_auth_username'] = "admin"
+default['private_chef']['opscode-erchef']['search_auth_password'] = "admin"
```
Comment (PrajaktaPurohit) [inline omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb:322]: The assumption here is that the user is manually created in the existing es setup. So this should not be a security concern.

File: omnibus/config/software/elasticsearch.rb [hunk 3]
```diff
@@ -35,6 +35,11 @@
          sha512: "bb02a5dc1caef97638a959ebba05dd649083c856334f30c670b851067292d7230e561d8759b15a80be73537d7a7efd9cef427d253cbb2efdbd6b168c6f9baa13"
 end
 
+version "7.9.1" do
```
Comment (PrajaktaPurohit) [inline omnibus/config/software/elasticsearch.rb:38]: We are not changing the default version of es with this change, so none of these change will work for internal es configuration.

File: omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb [hunk 4]
```diff
@@ -317,6 +317,9 @@
 default['private_chef']['opscode-erchef']['search_queue_mode'] = 'batch'
 default['private_chef']['opscode-erchef']['search_batch_max_size'] = '5000000'
 default['private_chef']['opscode-erchef']['search_batch_max_wait'] = '10'
+default['private_chef']['opscode-erchef']['search_auth_enabled'] = false
```
Comment (tas50) [inline omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb:320]: @jashaik @PrajaktaPurohit can we make sure to get a PR to the docs site to add these new config options

**Pull request merged**